src/serac/physics/materials/thermal_functional_material.hpp

```diff
@@ -39,52 +49,50 @@ class LinearIsotropicConductor {
                        "Specific heat capacity must be positive in the linear isotropic conductor material model.");
   }
 
-  /**
-   * @brief Function defining the thermal flux (constitutive response)
-   *
-   * @tparam T1 type of the temperature (e.g. tensor or dual type)
-   * @tparam T2 type of the temperature gradient (e.g. tensor or dual type)
-   * @param temperature_gradient Gradient of the temperature (du_dx)
-   * @return The thermal flux of the material model
-   */
-  template <typename T1, typename T2>
-  SERAC_HOST_DEVICE T2 operator()(const T1& /* temperature */, const T2& temperature_gradient) const
+  template <typename T1, typename T2, typename T3>
+  SERAC_HOST_DEVICE auto operator()(const T1&, const T2&, const T3& temperature_gradient) const
   {
-    return -1.0 * conductivity_ * temperature_gradient;
+    return MaterialResponse{.density                = density_,
+                            .specific_heat_capacity = specific_heat_capacity_,
+                            .heat_flux              = -conductivity_ * temperature_gradient};
   }
 
-  /**
-   * @brief The density (mass per volume) of the material model
-   *
-   * @tparam dim The dimension of the problem
-   * @return The density
-   */
-  template <int dim>
-  SERAC_HOST_DEVICE double density(const tensor<double, dim>& /* x */) const
+private:
+  /// Density
+  double density_;
+
+  /// Specific heat capacity
+  double specific_heat_capacity_;
+
+  /// Constant isotropic thermal conductivity
+  double conductivity_;
+};
+
+class ParameterizedLinearIsotropicConductor {
+public:
+  ParameterizedLinearIsotropicConductor(double density = 1.0, double specific_heat_capacity = 1.0,
+                                        double conductivity = 1.0)
+      : density_(density), specific_heat_capacity_(specific_heat_capacity), conductivity_(conductivity)
   {
-    return density_;
+    assert(conductivity > 0.0);
+    assert(density > 0.0);
+    assert(specific_heat_capacity > 0.0);
   }
 
-  /**
-   * @brief The specific heat capacity (heat capacity per unit mass) of the material model
-   *
-   * @tparam dim The dimension of the problem
-   * @tparam T Type of the temperature variable
-   */
-  template <typename T, int dim>
-  SERAC_HOST_DEVICE double specificHeatCapacity(const tensor<double, dim>& /* x */, const T& /* temperature */) const
+  template <typename T1, typename T2, typename T3, typename T4, typename T5>
+  SERAC_HOST_DEVICE auto operator()(const T1&, const T2&, const T3& temperature_gradient, const T4& parameter_1,
+                                    const T5& parameter_2) const
```

Comment:
Is T1 time and T2 temperature?

Reply:
Or I bet T1 is the location.

Reply:
Yep, `T1` is position. I'll go and add the placeholder comments to denote this.